contracts/services/generic-ibc-transfer/README.md

```diff
@@ -0,0 +1,78 @@
+# Valence Generic IBC Transfer service
+
+The **Valence Generic IBC Transfer** service allows to transfer funds over IBC from an **input account** on a source chain to an **output account** on a destination chain. It is typically used as part of a **Valence Workflow**. In that context, a **Processor** contract will be the main contract interacting with the Forwarder service.
+
+Note: this service should not be used on Neutron, which requires some fees to be paid to relayers for IBC transfers. For Neutron, prefer using the dedicated (and optimized) **Neutron IBC Transfer service** instead.
+
+## High-level flow
+
+```mermaid
+---
+title: Generic IBC Transfer Service
+---
+graph LR
+  IA((Input
+      Account))
+  OA((Output
+		  Account))
+  P[Processor]
+  S[Gen IBC Transfer
+    Service]
+  subgraph Chain 1
+  P -- 1/IbcTransfer --> S
+  S -- 2/Query balances --> IA
+  S -- 3/Do Send funds --> IA
+  end
+  subgraph Chain 2
+  IA -- 4/IBC Transfer --> OA
+  end
+```
+
+## Configuration
+
+The service is configured on instantiation via the `ServiceConfig` type.
+```rust
+struct ServiceConfig {
+  // Account from which the funds are pulled (on the source chain)
+  input_addr: ServiceAccountType,
+  // Account to which the funds are sent (on the destination chain)
+  output_addr: String,
+  // Denom of the token to transfer
+  denom: UncheckedDenom,
+  // Amount to be transferred, either a fixed amount or the whole available balance.
+  amount: IbcTransferAmount,
+  // Memo to be passed in the IBC transfer message.
+  memo: String,
+  // Information about the destination chain.
+  remote_chain_info: RemoteChainInfo,
+  // Denom map for the Packet-Forwarding Middleware, to perform a multi-hop transfer.
+  denom_to_pfm_map: BTreeMap<String, PacketForwardMiddlewareConfig>,
+}
+
+// Defines the amount to be transferred, either a fixed amount or the whole available balance.
+enum IbcTransferAmount {
+  // Transfer the full available balance of the input account.
+  FullAmount,
+  // Transfer the specified amount of tokens.
+  FixedAmount(Uint128),
+}
+
+pub struct RemoteChainInfo {
+  // Channel of the IBC connection to be used.
+  channel_id: String,
+  // Port of  the IBC connection to be used.
+  port_id: Option<String>,
```

Comment:
nit: extra space in comment.

btw, do we need port_id here? Is there any scenario where it won't be "transfer"? For what I've seen in the contract it's not being used anywhere so anything you pass here will be ignored.

Reply:
It's used by the Neutron service (cfr. [here](https://github.com/timewave-computer/valence-protocol/blob/3be7979a51b5ba22553f287e85a43de21e48793d/packages/ibc-utils/src/neutron.rs#L52)).. and they share the same types. I wanted to leave some flexibility in the config.

Reply:
Yeah I know it's used but I think that any value we put there that is not "transfer" will not work. In fact not sure why Neutron is exposing it in their custom proto type, either they know something that I don't or this is redundant